### DISCUSSION TITLE
Auto-save and Auto-load session in nvim

### DISCUSSION THREAD

**Discussion by atishdhu** [NvChad/NvChad#2203] [Q&A]
## Discussion
I am using `'rmagatti/auto-session'` plugin with the following config:

```lua
-- lua/custom/plugins.lua
{
    'rmagatti/auto-session',
    config = function()
      require("auto-session").setup {
        log_level = vim.log.levels.ERROR,
        auto_session_root_dir = '~/.config/nvim/session',
        auto_session_allowed_dirs = '~/.config/nvim/session',
        auto_session_enable_last_session = true,
        auto_session_enabled = true,
        auto_session_create_enabled = true,
        auto_save_enabled = true,
        auto_restore_enabled = true,
        auto_session_session_name = 'Session.vim', -- Set the session file name (optional)

        -- The following are already the default values, no need to provide them if these are already the settings you want.
        session_lens = {
          -- If load_on_setup is set to false, one needs to eventually call `require("auto-session").setup_session_lens()` if they want to use session-lens.
          load_on_setup = true,
          theme_conf = { border = true },
          previewer = false,
        },
      }
    end
  },
```

## Expected Behaviour
+ Open multiple buffers in nvim.
+ Quit nvim with `:q!`
+ Reopens nvim.
+ Last opened buffers before quitting nvim are loaded again.

## Actual Behaviour
+ No buffers are loaded.

**Comment by antoineco** [comment]
Add `event = "VimEnter",`.

Lazy plugins are loaded _lazily_ by default, so unless something explicitly requires it, it will likely remain unloaded.

**Reply by atishdhu** [reply]
I added the `event`:
```lua
{
    'rmagatti/auto-session',
    event = "VimEnter",
    config = function()
      require("auto-session").setup {
        ...
        ...
        ...
      }
    end,
  },
  ```

  but it does not work. I also tried adding `lazy = false` but not working either. Should this config be enough or is there additional configs needed to make it work.

**Reply by antoineco** [reply]
If NvChad did pick up the plugin and the plugin was loaded by Lazy on startup (you can check with the `:Lazy` command), but the plugin itself doesn't work, then the issue should be brought up at https://github.com/rmagatti/auto-session.

**Reply by atishdhu** [reply]
NvChad is picking up the plugin:
 
![image](https://github.com/NvChad/NvChad/assets/56167292/75e42b19-1966-4e30-a40e-33a9e11c778f)

I tried with a different plugin as well, NvChad is picking up the plugin in both cases but still does not work. I will try to ask in the plugins' community. Thankx :)

**Reply by siduck** [reply]
@atishdhu lazy=false should work!

**Reply by atishdhu** [reply]
I added a link to my [config](https://github.com/atishdhu/dotfiles/tree/main), if you could test it in a sandbox please let me know. I tried with `lazy = false` but still no result 😧

> ~/.config/nvim: https://github.com/atishdhu/dotfiles/tree/main

**Reply by atishdhu** [reply]
@antoineco @siduck It was a mistake from me when trying to open neovim. I was using `nvim .` but it should be only `nvim` without any arguments for the plugin to work. 🙂
